sodalite-server/src/main/java/org/icgc/dcc/sodalite/server/utils/JsonUtils.java

```diff
@@ -18,42 +18,87 @@
 package org.icgc.dcc.sodalite.server.utils;
 
 import java.io.IOException;
+import java.util.Map;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
 import com.fasterxml.jackson.databind.node.ObjectNode;
 import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
 import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
 import com.fasterxml.jackson.module.paramnames.ParameterNamesModule;
 
 import lombok.SneakyThrows;
+import lombok.val;
 
 /**
  * Utility functions related to deal with JSON
  */
 public class JsonUtils {
 
-  protected static final ObjectMapper mapper = new ObjectMapper().registerModule(new ParameterNamesModule())
-      .registerModule(new Jdk8Module())
-      .registerModule(new JavaTimeModule());
+  private static final String SINGLE_QUOTE = "'";
+  private static final String DOUBLE_QUOTE = "\"";
+
+  protected static final ObjectMapper mapper = mapper();
+
+  public static ObjectMapper mapper() {
+    val mapper = new ObjectMapper().registerModule(new ParameterNamesModule())
+        .registerModule(new Jdk8Module())
+        .registerModule(new JavaTimeModule());
+    mapper.disable(DeserializationFeature.FAIL_ON_MISSING_CREATOR_PROPERTIES);
+    mapper.disable(DeserializationFeature.FAIL_ON_UNRESOLVED_OBJECT_IDS);
+    mapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+    mapper.disable(DeserializationFeature.FAIL_ON_INVALID_SUBTYPE);
+    mapper.disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);
+
+    mapper.enable(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS);
+    return mapper;
+  }
 
   public static JsonNode readTree(String json) throws JsonProcessingException, IOException {
     return mapper.readTree(json);
   }
 
+  public static ObjectNode ObjectNode() {
+    return mapper.createObjectNode();
+  }
+
   @SneakyThrows
   public static String nodeToJSON(ObjectNode node) {
     return mapper.writeValueAsString(node);
   }
 
   @SneakyThrows
-  public static String nodeToJSON(JsonNode node) {
-    return mapper.writeValueAsString(node);
+  public static String toJson(Object o) {
+    return mapper.writeValueAsString(o);
   }
 
-  public static ObjectMapper mapper() {
-    return mapper;
+  @SneakyThrows
+  public static <T> T fromJson(String json, Class<T> toValue) {
+    return mapper.convertValue(mapper.readTree(json), toValue);
+  }
+
+  @SuppressWarnings("unchecked")
+  public static Map<String, Object> toMap(String json, String keyName) {
+    if (json == null || json.equals("")) {
+      json = "{}";
+    }
+    try {
+      return mapper.convertValue(mapper.readTree(json), Map.class);
+    } catch (IllegalArgumentException | IOException e) {
```

Comment:
Well, it was an attempt at error recovery... right now, any extra fields that are provided get automatically serialized into the "metadata" field as JSON data. 

However, it's possible that they might instead include an explicit "metadata" field, with, say, XML content; which would be invalid JSON; so I tried to handle that case by allowing a fallback where if the data we have isn't valid JSON, we make it valid, and serialize it so that people can get back what they put in...

Or we could just refuse to allow invalid JSON in the info field; or only allow them to input an "info" field, with
a single string that may/may not be valid JSON...

I was going to address that when I got into revising the validation code, which will have to change when we flatten the data structure.